Failed TestConnectVerify test on linux

https://github.com/projectdiscovery/naabu/runs/7556564016?check_suite_focus=true

```console
Run go test ./...
go: downloading github.com/stretchr/testify v1.8.0
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/pmezard/go-difflib v1.0.0
?   	github.com/projectdiscovery/naabu/v2/cmd/functional-test	[no test files]
?   	github.com/projectdiscovery/naabu/v2/cmd/internal/testutils	[no test files]
?   	github.com/projectdiscovery/naabu/v2/cmd/naabu	[no test files]
?   	github.com/projectdiscovery/naabu/v2/pkg/israce	[no test files]
?   	github.com/projectdiscovery/naabu/v2/pkg/privileges	[no test files]
ok  	github.com/projectdiscovery/naabu/v2/pkg/result	0.00[4](https://github.com/projectdiscovery/naabu/runs/7556564016?check_suite_focus=true#step:7:5)s
?   	github.com/projectdiscovery/naabu/v2/pkg/routing	[no test files]
ok  	github.com/projectdiscovery/naabu/v2/pkg/runner	0.[5](https://github.com/projectdiscovery/naabu/runs/7556564016?check_suite_focus=true#step:7:6)84s
--- FAIL: TestConnectVerify (0.01s)
    connect_test.go:31: 
        	Error Trace:	/home/runner/work/naabu/naabu/v2/pkg/scan/connect_test.go:31
        	Error:      	Not equal: 
        	            	expected: []int{1[7](https://github.com/projectdiscovery/naabu/runs/7556564016?check_suite_focus=true#step:7:8)[8](https://github.com/projectdiscovery/naabu/runs/7556564016?check_suite_focus=true#step:7:9)[9](https://github.com/projectdiscovery/naabu/runs/7556564016?check_suite_focus=true#step:7:10)5}
        	            	actual  : []int(nil)
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,2 @@
        	            	-([]int) (len=1) {
        	            	- (int) [17](https://github.com/projectdiscovery/naabu/runs/7556564016?check_suite_focus=true#step:7:18)895
        	            	-}
        	            	+([]int) <nil>
        	            	 
        	Test:       	TestConnectVerify
FAIL
FAIL	github.com/projectdiscovery/naabu/v2/pkg/scan	0.3[32](https://github.com/projectdiscovery/naabu/runs/7556564016?check_suite_focus=true#step:7:33)s
?   	github.com/projectdiscovery/naabu/v2/pkg/utils	[no test files]
FAIL
```

--- a/v2/go.mod
+++ b/v2/go.mod
@@ -8,7 +8,6 @@ require (
 	github.com/projectdiscovery/blackrock v0.0.0-20210903102120-5a9d2412d21d
 	github.com/projectdiscovery/cdncheck v0.0.4-0.20220322144854-b2d8ce308abb
 	github.com/projectdiscovery/clistats v0.0.8
-	github.com/projectdiscovery/dnsx v1.1.1-0.20220717105926-3bc1597b15c1
 	github.com/projectdiscovery/fdmax v0.0.3
 	github.com/projectdiscovery/fileutil v0.0.0-20220705195237-01becc2a8963
 	github.com/projectdiscovery/goflags v0.0.9
@@ -26,6 +25,7 @@ require (
 	github.com/logrusorgru/aurora v2.0.3+incompatible
 	github.com/miekg/dns v1.1.50
 	github.com/pkg/errors v0.9.1
+	github.com/projectdiscovery/dnsx v1.1.1-0.20220726121728-699bed01313a
 	github.com/projectdiscovery/ipranger v0.0.3-0.20220715104731-7638578d1725
 	github.com/projectdiscovery/mapcidr v1.0.1
 	github.com/projectdiscovery/retryablehttp-go v1.0.2
@@ -67,7 +67,7 @@ require (
 	github.com/modern-go/reflect2 v1.0.2 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/projectdiscovery/hmap v0.0.2 // indirect
-	github.com/projectdiscovery/retryabledns v1.0.14-0.20220717105827-2d0b0e81dd89 // indirect
+	github.com/projectdiscovery/retryabledns v1.0.14 // indirect
 	github.com/saintfish/chardet v0.0.0-20120816061221-3af4cd4741ca // indirect
 	github.com/syndtr/goleveldb v1.0.0 // indirect
 	github.com/yl2chen/cidranger v1.0.2 // indirect

--- a/v2/pkg/runner/util_test.go
+++ b/v2/pkg/runner/util_test.go
@@ -40,7 +40,7 @@ func Test_host2ips(t *testing.T) {
 			}
 			assert.Equal(t, tt.want, got)
 			// As some distributions don't handle correctly ipv6 we compare results only if necessary
-			if len(gotV6) > 0 && len(gotV6) > 0 {
+			if len(gotV6) > 0 && len(tt.wantV6) > 0 {
 				assert.Equal(t, tt.wantV6, gotV6)
 			}
 		})

--- a/v2/pkg/runner/util_test.go
+++ b/v2/pkg/runner/util_test.go
@@ -20,7 +20,7 @@ func Test_host2ips(t *testing.T) {
 		{"10.10.10.0/24", nil, nil, true},
 	}
 
-	r, err := NewRunner(&Options{IPVersion: []string{"4", "6"}})
+	r, err := NewRunner(&Options{IPVersion: []string{"4", "6"}, Retries: 1})
 	assert.Nil(t, err)
 	if dnsclient, err := dnsx.New(dnsx.DefaultOptions); err != nil {
 		assert.Error(t, err)

--- a/v2/pkg/runner/util_test.go
+++ b/v2/pkg/runner/util_test.go
@@ -15,7 +15,7 @@ func Test_host2ips(t *testing.T) {
 		wantErr bool
 	}{
 		{"10.10.10.10", []string{"10.10.10.10"}, nil, false},
-		{"localhost", []string{"127.0.0.1"}, nil, false}, // ubuntu doesn't hardcode ::1
+		{"localhost", []string{"127.0.0.1"}, []string{"::1"}, false}, // some linux distribution don't have ::1 in /etc/hosts
 		{"aaaa", nil, nil, true},
 		{"10.10.10.0/24", nil, nil, true},
 	}
@@ -39,7 +39,10 @@ func Test_host2ips(t *testing.T) {
 				assert.Nil(t, err)
 			}
 			assert.Equal(t, tt.want, got)
-			assert.Equal(t, tt.wantV6, gotV6)
+			// As some distributions don't handle correctly ipv6 we compare results only if necessary
+			if len(gotV6) > 0 && len(gotV6) > 0 {
+				assert.Equal(t, tt.wantV6, gotV6)
+			}
 		})
 	}
 }

--- a/v2/pkg/runner/util_test.go
+++ b/v2/pkg/runner/util_test.go
@@ -15,7 +15,7 @@ func Test_host2ips(t *testing.T) {
 		wantErr bool
 	}{
 		{"10.10.10.10", []string{"10.10.10.10"}, nil, false},
-		{"localhost", []string{"127.0.0.1"}, []string{"::1"}, false},
+		{"localhost", []string{"127.0.0.1"}, nil, false}, // ubuntu doesn't hardcode ::1
 		{"aaaa", nil, nil, true},
 		{"10.10.10.0/24", nil, nil, true},
 	}
@@ -40,7 +40,6 @@ func Test_host2ips(t *testing.T) {
 			}
 			assert.Equal(t, tt.want, got)
 			assert.Equal(t, tt.wantV6, gotV6)
-
 		})
 	}
 }

